src/components/virtualRepeat/virtualRepeater.js

```diff
@@ -131,11 +139,40 @@ VirtualRepeatContainerController.prototype.getScrollSize = function() {
 /**
  * Sets the scrollHeight or scrollWidth. Called by the repeater based on
  * its item count and item size.
- * @param {number} The new size.
+ * @param {number} size The new size.
  */
 VirtualRepeatContainerController.prototype.setScrollSize = function(size) {
   if (this.scrollSize !== size) {
-    this.sizer.style[this.isHorizontal() ? 'width' : 'height'] = size + 'px';
+    var dimension =  this.isHorizontal() ? 'width' : 'height';
+    var crossDimension = this.isHorizontal() ? 'height' : 'width';
+
+    // If the size falls within the browser's maximum explicit size for a single element, we can
+    // set the size and be done. Otherwise, we have to create children that add up the the desired
+    // size.
+    if (size < MAX_ELEMENT_SIZE) {
+      this.sizer.style[dimension] = size + 'px';
+    } else {
+      // Clear any existing dimensions.
+      this.sizer.innerHTML = '';
+      this.sizer.style[dimension] = 'auto';
+      this.sizer.style[crossDimension] = 'auto';
+
+      // Divide the total size we have to render into N max-size pieces.
+      var numChildren = Math.floor(size / MAX_ELEMENT_SIZE);
+
+      // Element template to clone for each max-size piece.
+      var sizerChild = document.createElement('div');
+      sizerChild.style[dimension] = MAX_ELEMENT_SIZE + 'px';
+      sizerChild.style[crossDimension] = '1px';
+
+      for (var i = 0; i < numChildren; i++) {
+        this.sizer.appendChild(sizerChild.cloneNode(false));
```

Comment:
I considered it, but the modern browsers shouldn't care since we're not reading any layout properties while we're appending.